modules/src/mock_context.rs

```diff
@@ -59,16 +68,31 @@ impl MockContext {
     /// to this client a mock client state and a mock consensus state for height `height`. The type
     /// of this client is implicitly assumed to be Mock.
     pub fn with_client(self, client_id: &ClientId, height: Height) -> Self {
+        self.with_client_parametrized(client_id, height, Some(ClientType::Mock), Some(height))
+    }
+
+    /// Similar to `with_client`, this function associates a client record to this context, but
+    /// additionally permits to parametrize two details of the client. If `client_type` is None,
+    /// then the client will have type Mock, otherwise the specified type. If
+    /// `consensus_state_height` is None, then the client will be initialized with a consensus
+    /// state matching the same height as the client state (`client_state_height`).
+    pub fn with_client_parametrized(
+        self,
+        client_id: &ClientId,
+        client_state_height: Height,
+        client_type: Option<ClientType>,
+        consensus_state_height: Option<Height>,
```

Comment:
Still not sure about this, seems that the `client_state_height` is a bit redundant, not clear what it means if the two heights are different. Maybe just pass an `AnyHeader`. But let's leave this for the next step where we support tendermint client and consensus states in the mock context and things will become more clear.

Reply:
Will merge this after #309. Thanks Anca!